### ISSUE TITLE
declare_covariates should throw error with nonsense lower_units_per_level = 

### ISSUE THREAD

**Issue by acoppock** [egap/registration#13]

**Comment by jaspercooper** [comment]
Definitely, just haven't had the time to get to it yet. I need to to checks on the lower_units_per_level argument and on the N_per_level argument, and also create some flexible way of creating these endogenously if the user doesn't supply them. Will get to work on it.
